brili.ts

```diff
@@ -754,18 +754,18 @@ function evalInstr(instr: bril.Instruction, state: State): Action {
   }
 
   case "char2int": {
-    let code = getChar(instr, state.env, 0).charCodeAt(0);
-    let val = BigInt.asIntN(64, BigInt(code));
+    let code = getChar(instr, state.env, 0).codePointAt(0);
+    let val = BigInt.asIntN(64, BigInt(code as number));
     state.env.set(instr.dest, val);
     return NEXT;
   }
 
   case "int2char": {
-    let i = getInt(instr, state.env, 0)
-    if(i > 65535 || i < 0) {
+    let i = getInt(instr, state.env, 0);
+    if(i > 1114111 || i < 0) {
```

Comment:
I just want to double check if this range is sufficient to rule out "lone surrogates". I'm just guessing based purely off of comparing against how a rust `char` is defined so I'm not 100%. Note how there is a gap in Rust's char range where the surrogate values are: https://doc.rust-lang.org/std/primitive.char.html#validity

Reply:
I was searching more, and since [fromCodePoint](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCodePoint#compared_to_fromcharcode) is according to the UTF-32 code unit, I guess an integer in the surrogate range will return an [ill-formed](https://www.unicode.org/versions/Unicode15.0.0/ch03.pdf#page=54) string. It is still a valid code point though, that's why the `fromCodePoint` won't throw with the surrogate range.

I think we can prevent that.